docs/notebooks/openai_gym_lunar_lander.pct.py

```diff
@@ -0,0 +1,244 @@
+#
+# ### Optimization problem
+#
+# Now let's see how this task can be formulated as an optimization problem. We will be following an approach used by Turbo <cite data-cite="eriksson2019scalable"/> and BOSH <cite data-cite="Moss2020BOSHBO"/> papers. The environment comes with a heuristic controller that makes decisions based on the current position and velocity of the module. This controller can be parametrized by turning all numerical values in it into parameters. These parameters form our optimization search space. The objective is the same as in the original RL setup: maximize the reward. Therefore we will be using Trieste to learn how to land the module safely on the designated pad, without taking too much time and wasting too much fuel.
+#
+# The original code for the heuristic controller can be found in [OpenAI Gym GitHub repo](https://github.com/openai/gym/blob/master/gym/envs/box2d/lunar_lander.py). Here is the parametrized version, taken from the [repository](https://github.com/uber-research/TuRBO) of the Turbo paper:
+
+# %%
+# controller code is copied verbatim from https://github.com/uber-research/TuRBO
+def heuristic_Controller(s, w):
+    angle_targ = s[0] * w[0] + s[2] * w[1]
+    if angle_targ > w[2]:
+        angle_targ = w[2]
+    if angle_targ < -w[2]:
+        angle_targ = -w[2]
+    hover_targ = w[3] * np.abs(s[0])
+
+    angle_todo = (angle_targ - s[4]) * w[4] - (s[5]) * w[5]
+    hover_todo = (hover_targ - s[1]) * w[6] - (s[3]) * w[7]
+
+    if s[6] or s[7]:
+        angle_todo = w[8]
+        hover_todo = -(s[3]) * w[9]
+
+    a = 0
+    if hover_todo > np.abs(angle_todo) and hover_todo > w[10]:
+        a = 2
+    elif angle_todo < -w[11]:
+        a = 3
+    elif angle_todo > +w[11]:
+        a = 1
+    return a
+
+
+
+steps_limit = 1000
+timeout_reward = -100
+
+
+def demo_heuristic_lander(env, w, print_reward=False):
+    total_reward = 0
+    steps = 0
+    s = env.reset()
+
+    while True:
+        if steps > steps_limit:
+            total_reward -= timeout_reward
+            break
+
+        a = heuristic_Controller(s, w)
+        s, r, done, info = env.step(a)
+        total_reward += r
+
+        steps += 1
+        if done:
+            break
+
+    if print_reward:
+        print(f"Total reward: {total_reward}")
+
+    return total_reward
+
+
+# %% [markdown]
+# Original values for the various parameters take values between 0.05 and 1.0. We will set the search range for each parameter to be the same, from 0.0 to 1.2.
```

Comment:
is this taken from TurBO? or it follows our own logic? are all of those parameters rescaled, so it makes sense to have them all within [0, 1.2]? 
Feels like this sentence says either too much or too little.

Reply:
I see your confusion, and tried to update the sentence. Let me know if that's not clear still.